File: DSA/DSA03013 SẮP ĐẶT XÂU KÝ TỰ 2.cpp
#include <bits/stdc++.h>
using namespace std;

void submit()
{
    int d;
    string s;
    cin >> d >> s;

    vector<int> alpha(127);
    int max_num = 0;
    for (auto &&i : s)
    {
        alpha[i]++;
        max_num = max(max_num, alpha[i]);
    }
    int max = (s.length()+1)/d;

    max_num <= max ? (cout << 1) : (cout << -1);
    
    cout << endl;
}

int main()
{
    ios::sync_with_stdio(false);                                    // Speed up input/output
    cin.tie(nullptr);
    cout.tie(nullptr);

    int test;
    cin >> test;
    while (test--)
    {
        submit();
    }
    

    return 0;
}

File: DSA_python/DSA03013 SẮP ĐẶT XÂU KÝ TỰ 2.py
def submit():
    d = int(input())
    s = input()
    alpha = [0]*127
    maxx = 0

    for i in range(len(s)):
        alpha[ord(s[i])] += 1
        maxx = max(maxx, alpha[ord(s[i])])
    maxi = (len(s)+1)/d

    print(1 if maxx <= maxi else -1)

test = int(input())
for i in range(test):
    submit()